feat: add contributing guide (#175)
FuelLabs/docs-hub: docs/contributing/general.mdx

---
title: General
category: Contributing
parent:
  label: All Docs
  link: /
---

# General

You can find the repository for this website on [GitHub](https://github.com/FuelLabs/docs-hub/tree/master). For instructions on how to run the repository locally, see the [README](https://github.com/FuelLabs/docs-hub/blob/master/README.md).

## Contribution flow

This is a rough outline of what a contributor's workflow looks like:

> - Create a git branch from where you want to base your work. This is usually master.

Reviewer: This means they should create a feature branch, right? Should it say something like:

```suggestion
- Create a feature branch off of the master branch, which is typically the base for your work.
```

- Make your changes, and commit your work.
- Run tests and make sure all tests pass.
- Push your changes to a branch in your fork of the repository and submit a pull request.
  - Use one of the following tags in the title of your PR:
    - `feat:` - A new feature
    - `fix:` - A bug fix
    - `docs:` - Documentation only changes
    - `style:` - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
    - `refactor:` - A code change that neither fixes a bug nor adds a feature
    - `perf:` - A code change that improves performance
    - `test:` - Adding missing tests or correcting existing tests
    - `build:` - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
    - `ci:` - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
    - `chore:` - Other changes that don't modify `src` or test files
    - `revert:` - Reverts a previous commit
- Complete the contributor agreement on the PR if it is not already completed.
- Your PR will be reviewed and some changes may be requested.

>   - Once you've made changes, your PR must be re-reviewed and approved.

Reviewer: ```suggestion
  - Once you've made the requested changes, your PR must be re-reviewed and approved.
```
